daemon/cmd/cells.go

```diff
@@ -90,5 +94,9 @@ var (
 			newWireguardAgent,
 			newDatapath,
 		),
+
+		cell.Provide(func(dp datapath.Datapath) ipcacheTypes.NodeHandler {
```

Comment:
I'd keep adaptions like this next to where they're used, e.g:

```
cell.Provide(
  func(dp datapath.Datapath) ipcacheTypes.NodeHandler { return dp.Node() },
  newPolicyMess,
)

Though here it would seem ok to me to just take `datapath.Datapath` in `newPolicyMess`.